src/options/optionsMain.ts

```diff
@@ -115,6 +115,11 @@ export async function getOptions(rawOptions: any): Promise<AppOptions> {
     log.setLevel('info');
   }
 
+  if (rawOptions.wvvmp) {
+      process.env.ELECTRON_MIRROR = 'https://github.com/castlabs/electron-releases/releases/download/';
+      options.packager.electronVersion = '11.0.0-wvvmp';
```

Comment:
Rather than override the version, what about building the wvvmp version from the actually-requested version (or default if not passed), appending `-wvvmp`.

Something like

```ts
const wvvmpString = rawOptions.wvvmp ? '-wvvmp' : '';
options.packager.electronVersion = ${electronVersion}${wvvmpString}`
```

This way, a call becomes `nativefier -e 11.0.1 -wvvmp netflix.com`, keeping requested version and wvvmp-enablement orthogonal.

Reply:
I think this will achieve what you want:
```ts
if (rawOptions.wvvmp) {
  process.env.ELECTRON_MIRROR = 'https://github.com/castlabs/electron-releases/releases/download/';
  const electronVersion = options.packager.electronVersion;
  options.packager.electronVersion = `${electronVersion}-wvvmp`;
```

But I'll need to do some more testing, castLabs does not always release a wvvmp enabled version of Electron.

Follow on: Perhaps `if(rawOptions.wvvmp){}` should fall under `if(rawOptions.electronVersion){}` so that rawOptions.electronVersion can be tested before using it.

Reply:
> so that rawOptions.electronVersion can be tested before using it.

Indeed, there's be a small refactoring/reordering to do what I'd like here; go ahead.

> I'll need to do some more testing, castLabs does not always release a wvvmp enabled version of Electron.

Sure, more error handling welcome if you wish to invest in it but don't overdo it; I'd be fine leaving upon users to know this is "bleeding edge". In the extra console.log I was asking to warn that "you're on your own here", maybe mention what you're saying here: that users should pick a version for which a build is available at https://github.com/castlabs/electron-releases/releases , and retry with another version if build fails.

Reply:
Ok, so I may have gone off the rails here... but if you include remote-git-tags then we can test user input against the castLabs repo.

```ts
const remoteGitTags = require('remote-git-tags');

    if (rawOptions.wvvmp) {
        process.env.ELECTRON_MIRROR = 'https://github.com/castlabs/electron-releases/releases/download/';
	const electronVersion = options.packager.electronVersion;
        options.packager.electronVersion = `${electronVersion}-wvvmp`;
	const tags = (await remoteGitTags('https://github.com/castlabs/electron-releases'));
        if (tags.get(`v${electronVersion}-wvvmp`) === undefined) {
            throw `\nERROR: castLabs Electron version "${options.packager.electronVersion}" does not exist. \nVerify versions at https://github.com/castlabs/electron-releases/releases. \nAborting.`;
        }
        log.warn(`\nATTENTION: Using the **unofficial** Electron for Content Security from castLabs`, "\nIt implements Google's Widevine Content Decryption Module (CDM) for DRM-enabled playback.", "\nBe sure to select a valid version from https://github.com/castlabs/electron-releases/releases !", `\nSimply abort & re-run without passing the wvvmp flag to default to ${constants_1.DEFAULT_ELECTRON_VERSION}`);
    }
```

Reply:
>  if you include remote-git-tags then we can test user input against the castLabs repo.

@sorhtyre what about avoiding the additional dependency and instead asserting that a test HTTP call to https://github.com/castlabs/electron-releases/releases/tag/vVERSION-wvvmp doesn't 404? If 👍 to you, use `axios` for the HTTP call, which we already depend on.

Reply:
Boom!

```ts
const resourceCheck = require('axios');

  await resourceCheck.get(`https://github.com/castlabs/electron-releases/releases/tag/v${electronVersion}-wvvmp`)
    .catch(function (error) {
      throw `\nERROR: castLabs Electron version "${options.packager.electronVersion}" does not exist. \nVerify versions at https://github.com/castlabs/electron-releases/releases. \nAborting.`;
    });
```

Reply:
Great! A few things:

1. The import should be at the top of the file, with the other imports (ordered alphabetically), bear the name of the library, and use static `import` rather than dynamic `require`:

```ts
import * as axios from 'axios';
```

Then, don't mix async/await style with promise style. A promise `.catch` translates to a `catch` clause in async/await world, so you can:

```ts
  try {
    await axios.get(`https://github.com/castlabs/electron-releases/releases/tag/v${electronVersion}-wvvmp`)
  } catch (error) {
    throw `\nERROR: castLabs Electron version "${options.packager.electronVersion}" does not exist. \nVerify versions at https://github.com/castlabs/electron-releases/releases. \nAborting.`;
  }

Reply:
:+1: 
Sorry about that, I was posting the .js version of the code rather than the .ts version. As for the `.catch` vs `catch` that happened because I don't actually know this language and am learning this all on the fly...

Reply:
> Sorry about that, I was posting the .js version of the code rather than the .ts version. As for the `.catch` vs `catch` that happened because I don't actually know this language and am learning this all on the fly...

@sorhtyre zeeeero worry 🙂! Only helping you get your PR ready for merging. You're doing great!

Reply:
👍